pkg/agent/plugin/workloadattestor/k8s/k8s_test.go

```diff
@@ -175,7 +205,8 @@ func TestK8s_AttestPidNotInPodAfterRetry(t *testing.T) {
 	req := workloadattestor.AttestRequest{Pid: int32(pid)}
 	resp, err := plugin.Attest(ctx, &req)
 	require.Error(t, err)
-	require.Empty(t, resp.Selectors)
```

Comment:
This is actually a non-trivial API change. Instead of `resp.Selectors` being empty, `resp` is now `nil`. Do we think this behavior is relied on anywhere? If so, we may re-think changing it.

Reply:
gRPC returns a nil response if an error is returned. Our built-in plugin pipe has the same behavior. I think we're ok here.